metrics/coverage_converter/docs/01_coverage_converter_design.md

```diff
@@ -0,0 +1,148 @@
+
+Importing CI builds coverage data into the Metrics application could be tricky as there are many coverage formats and tools like Instanbul, LCOV, etc... To simplify integration we need to create a tool that can be used to convert coverage data from specific coverage tool output format into Metrics coverage format - the Coverage Converter tool. To make the Coverage Converter tool architecture clean and understandable, we need to create a document that will explain the main components and principles of the Coverage tool.
+
+# Goals
+
+> Identify success metrics and measurable goals.
+
+- Explain the purpose of the Coverage Converter tool.
+- Explain the main components of the Coverage Converter tool.
+- Explain adding the new specific format converter to the Coverage Converter tool.
+
+# Non-Goals
+
+> Identify what's not in scope.
+
+The document does not explain and shows the implementation details.
+
+# Design
+
+> Explain and diagram the technical design
+
+## Main interfaces and classes
+
+The main purpose of this tool is to convert the given supported coverage report to the [CI integrations format](https://github.com/platform-platform/monorepo/blob/codecov_converter_design/metrics/ci_integrations/docs/01_ci_integration_module_architecture.md#coverage-importing). To do so, we should create the following interfaces and abstract classes:
+
+- The `CoverageConverterCommand` interface - the common code coverage command interface. 
+- The `CoverageConverter` interface that will represent the common interface for converting the specific coverage report format to the `CI integrations` coverage report format.
+
+Let's consider the class diagram explaining the main interfaces and dependencies between them: 
+
+![Interfaces diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/platform-platform/monorepo/codecov_converter_design/metrics/coverage_converter/docs/diagrams/coverage_interfaces_diagram.puml)
+
+As we can see above, the `CoverageConverterCommand` uses the `CoverageConverter` to convert the specific coverage report to the `CoverageModel` class that represents the `CI integrations` coverage report.
+
+Let's consider the activity diagram of the coverage conversion process: 
+
+![Coverage conversion diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/codecov_converter_design/metrics/coverage_converter/docs/diagrams/coverage_conversion_activity.puml)
+
+To run the command line application, we should have a `CommandRunner` class. Let's name it `CoverageConverterRunner`. This class will extend `CommandRunner`. It will be the place where we'll add our specific `CoverageConverterCommand`s.
+
+Also, we should be able to pass the following arguments to the Coverage Converter tool: 
+
+- input - the file from where we'll read the specific coverage report;
+- output - the file to which we'll write the formatted coverage report;
```

Comment:
As we have a specific file name used for coverage artifact should we really provide the ability to specify output? Or we want to make sure that if this parameter is not provided - the default name will be used instead?